### ISSUE TITLE
Lack of Proper Documentation

### ISSUE THREAD

**Issue by subinsebastien** [satyan/sugar#386]
I'm one of among the many who follows this project for a long time. With the recent beta release v1.4, things tend to break if we use, now obsolete v1.3 compatible code. In general, I see a lack of effort put into documenting the project properly. Compared to similar solutions, Sugar does not have professional documentation for any of the releases, and in many cases even lacks release notes. The original documentation is so pure and vanilla, which does not cover many use cases. I generally see a lot of issues tickets being opened for plain questions on how to do certain things with Sugar, which is not the ideal use case of `Github Issues` as such. I personally tried to document the project and planned to send pull requests, but I find even myself don't know how to use Sugar for certain requirements. While the project gets updates only 2 or 3 months, it doesn't take much effort to maintain a descriptive documentation, in my opinion.

 -- Feel free to close this issue if you find it OT

**Comment by subinsebastien** [comment]
Issue that is not actually an issue #377

**Comment by subinsebastien** [comment]
#375 How to turn off Sugar logging

**Comment by subinsebastien** [comment]
#372 Is there event after query finishes or database connection closed ?

**Comment by subinsebastien** [comment]
#371 is it possible not to put null as String into String columns?

**Comment by subinsebastien** [comment]
#353 How can I do an sql join between tables?

**Comment by clay-to-n** [comment]
Agreed on this.  Upgrading to 1.4 from 1.3 is breaking, which means Sugar isn't following semantic versioning correctly - and there isn't an upgrade guide, so we don't know where to start besides looking through commits and issues.

This is a great library, but proper documentation is a necessity.

**Comment by sibeliusseraphini** [comment]
@clay-to-n I've updated the docs, and I agree with you about the semantic versioning that's wrong for this library
